File: frontend/src/features/sensors/interfaces/SensorProcessed.ts
export interface SensorProcessedColumn {
  value: number[];
  iqr_lower: number | null;
  iqr_upper: number | null;
}

export interface SensorProcessed {
  count: number;
  temperature: SensorProcessedColumn;
  humidity: SensorProcessedColumn;
  air_quality: SensorProcessedColumn;
}

File: backend/apps/sensors/serializers/sensor_processed.py
from rest_framework import serializers


class ResponseSensorProcessedDataSerializer(serializers.Serializer):
    value = serializers.ListField(
        child=serializers.FloatField(), read_only=True
    )
    iqr_lower = serializers.FloatField(allow_null=True, read_only=True)
    iqr_upper = serializers.FloatField(allow_null=True, read_only=True)


class ResponseSensorProcessedSerializer(serializers.Serializer):
    count = serializers.IntegerField(read_only=True)
    temperature = ResponseSensorProcessedDataSerializer()
    humidity = ResponseSensorProcessedDataSerializer()
    air_quality = ResponseSensorProcessedDataSerializer()
    timestamp = serializers.ListField(
        child=serializers.DateTimeField(read_only=True)
    )
